File: src/main/java/com/kerimsenturk/labreport/dto/request/CreateDiseaseRequest.java
package com.kerimsenturk.labreport.dto.request;

import com.kerimsenturk.labreport.dto.validator.HospitalPersonalIdValid;
import com.kerimsenturk.labreport.dto.validator.PatientIdValid;
import com.kerimsenturk.labreport.model.enums.LabRequestType;


public record CreateDiseaseRequest(
        @PatientIdValid
        String patientId,
        @HospitalPersonalIdValid
        String doctorId,
        LabRequestType labRequestType) {
}

File: src/main/web-ui/src/domain/payload/request/CreateDiseaseRequest.ts
import {LabRequestType} from "../../model/Disease";

export class CreateDiseaseRequest {
    patientId: string;

    doctorId: string;

    labRequestType: LabRequestType;

    constructor(patientId: string, doctorId: string, labRequestType: LabRequestType) {
        this.patientId = patientId;
        this.doctorId = doctorId;
        this.labRequestType = labRequestType;
    }
}

